File: seeker/snippet/linked_list.py
#date: 2022-08-25T16:53:18Z
#url: https://api.github.com/gists/4540fb821d21adee5df29609aec131bb
#owner: https://api.github.com/users/jermspeaks

# Node for LinkedList
class Node:
  def __init__(self, value):
    self.value = value
    self.next = null

# A data structure that is a series of nodes,
# and each node points to the next node of the list

# Linked lists use the “last-in-first-out” method (similar to a stack)where nodes are added to and deleted from the same end.
class LinkedList:
    def __init__(self):
        self.head = null
        self.tail = null
        self.length = 0

    def isEmpty():
        # Check whether the queue is empty
        self.length == 0

    def push(value):
        # Add an element to the linked list
        new_node = Node(value)
        if not self.head:
            self.head = new_node
            self.tail = new_node
        else:
            self.tail.next = new_node
            self.tail = new_node 
        self.length += 1        

    def pop():
        self
        # Remove an element from the linked list  

  

    def get(index): 
        # takes an index and returns the node at that index
        self

    def delete(index): 
        # takes an index and removes node at that index
        self

    def print():
        currNode = self.head
        while (currNode != None):
            currNode = currNode.next

# list = new LinkedList()
# list.push("Emma");
# list.push("Sarah");
# list.push("Ivy");

File: seeker/snippet/LinkedList.go
//date: 2023-04-11T16:48:25Z
//url: https://api.github.com/gists/caa93e0f10708cd79e408ccaa2d64084
//owner: https://api.github.com/users/NoobforAl

package main

import (
	"fmt"

	"github.com/kr/pretty"
)

type Value interface {
	int | int64 | int32 | int16 | int8 |
		float64 | float32 | byte | string
}

type LinkedList[T Value] struct {
	Val  T
	Next *LinkedList[T]
}

func (list *LinkedList[T]) Insert(Val T) {
	for tmp := list; tmp != nil; tmp = tmp.Next {
		if tmp.Next == nil {
			tmp.Next = &LinkedList[T]{Val, nil}
			break
		}
	}
}

func (list *LinkedList[T]) Pop() {
	for tmp := list; tmp.Next != nil; tmp = tmp.Next {
		if tmp.Next.Next == nil {
			tmp.Next = nil
			break
		}
	}
}

func (list *LinkedList[T]) Find(Val T) *LinkedList[T] {
	for tmp := list; tmp != nil; tmp = tmp.Next {
		if tmp.Val == Val {
			return tmp
		}
	}
	return nil
}

func main() {
	// Make Head
	list := LinkedList[int]{1, nil}

	// Insert 2 Value
	list.Insert(2)
	list.Insert(3)

	// Pop 2 Value
	list.Pop()

	// Find One Value
	f := list.Find(1)
	fmt.Printf("Find: %# v\n", pretty.Formatter(f))

	fmt.Printf("All Value: %# v", pretty.Formatter(list))
}
